web/src/main/java/com/epam/esm/controller/CertificateRESTController.java

```diff
@@ -0,0 +1,139 @@
+        Link prevLink = linkTo(methodOn(CertificateRESTController.class).listAllCertificates(Pagination.findPrevPage(page), size))
+                .withRel("prev");
+        Link nextLink = linkTo(methodOn(CertificateRESTController.class).listAllCertificates(Pagination.findNextPage(page, lastPage), size))
+                .withRel("next");
+        List<ModelCertificate> models = service.getAllEntity(size, offset);
+        return CollectionModel.of(models, prevLink, nextLink);
+    }
+
+    /**
+     * get certificate by id
+     *
+     * @param id the id
+     * @return certificate
+     */
+    @GetMapping("/certificates/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public Optional<ModelCertificate> getCertificateById(@PathVariable long id) {
+        Optional<ModelCertificate> model = Optional.ofNullable(service.getEntity(id)).get();
+        model.get().add(linkTo(methodOn(CertificateRESTController.class)
+                .listAllCertificates(1,5))
+                .withRel("certificate")
+                .withType(HttpMethod.GET.name()));;
+        return model;
+    }
+
+    /**
+     * update certificate
+     * enter the fields that you want to update
+     * date not updated except the date
+     *
+     * @param certificate the certificate
+     * @param id          the id
+     * @return updated certificate
+     */
+    @PatchMapping("/certificates/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public Certificate updateCertificate(@RequestBody OnlyCertificate certificate, @PathVariable long id) {
+        service.updateEntity(id, certificate);
+        return certificateMapper.mapFrom(certificate);
+    }
+
+    /**
+     * delete certificate by id
+     *
+     * @param id the id
+     * @return string response
+     */
+    @DeleteMapping("/certificates/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public String deleteCertificate(@PathVariable long id) {
+        service.deleteEntity(id);
+        return "Certificate with ID = " + id + " was deleted.";
+    }
+
+    /**
+     * @param name the name
+     * @return certificate by name or part of name
+     */
+    @GetMapping("/certificates/name/{name}")
+    @ResponseStatus(HttpStatus.OK)
+    public List<ModelCertificate> getCertificateByName(@PathVariable String name) {
+        return service.getCertificateByName(name);
+    }
+
+    /**
+     * @param tagName the tagName
+     * @return certificate by tagName
+     */
+    @GetMapping("/certificates/tag/{tagName}")
+    @ResponseStatus(HttpStatus.OK)
+    public List<ModelCertificate> getCertificateByTag(@PathVariable String tagName) {
+        return service.getCertificatesByTag(tagName);
+    }
+
+    @GetMapping("/certificates/tags")
+    public List<ModelCertificate> getCertificatesByTags(@RequestParam String tagName1,
+                                                        @RequestParam String tagName2) {
+        return service.getCertificatesByTags(tagName1, tagName2);
```

Comment:
I've already typed the example of getting certificates by params. It should handle any count of tags, not only 2

Reply:
made